new-log-viewer/src/components/MenuBar/ExportLogsButton.tsx

```diff
@@ -0,0 +1,57 @@
+import {useContext} from "react";
+
+import {
+    CircularProgress,
+    Typography,
+} from "@mui/joy";
+
+import DownloadIcon from "@mui/icons-material/Download";
+
+import {StateContext} from "../../contexts/StateContextProvider";
+import {
+    EXPORT_LOG_PROGRESS_COMPLETE,
+    EXPORT_LOG_PROGRESS_INITIALIZATION,
+} from "../../services/LogExportManager";
+import SmallIconButton from "./SmallIconButton";
+
+
+/**
+ * Represents a button for triggering log exports and displays the progress.
+ *
+ * @return
+ */
+const ExportLogsButton = () => {
+    const {exportLogs, exportProgress} = useContext(StateContext);
+    const handleExportLogsButtonClick = () => {
+        exportLogs();
+    };
+
+    return (
+        <SmallIconButton
+            disabled={null !== exportProgress && 1 !== exportProgress}
```

Comment:
```
        if (this.#chunks.length === this.#numChunks) {
            this.#download();
            this.#chunks.length = 0;

            return EXPORT_LOG_PROGRESS_COMPLETE;
        }
```

I think `1 === exportProgress` wouldn't be true until the file has been `download()`ed. If we're referring to the time between `EXPORT_LOG_PROGRESS_COMPLETE` is returned and the browser finishes downloading the file, I'm not too sure if there're any APIs that can be used to check the download progress.

Reply:
So there are two cases I tested which generate console errors:

* When no file is loaded in the log viewer.
* When I drag a file in, before it finishes loading.

Since we're printing console errors, I think we definitely have a way of detecting those two cases. Just need to propagate the conditions here somehow.

Reply:
Gotcha. Essentially, all actions should be disabled until a file is loaded by the log-viewer. Do you think we can postpone this to another PR?

Reply:
Is it not sufficient to add a check if `fileName` is null?

Reply:
Sure